src/ucs/sys/sock.c

```diff
@@ -366,14 +370,33 @@ ucs_status_t ucs_socket_server_init(const struct sockaddr *saddr, socklen_t sock
         goto err_close_socket;
     }
 
-    status = ucs_sockaddr_get_port(saddr, &port);
-    if (status != UCS_OK) {
-        goto err_close_socket;
+    if ((ports_range_start != 0) && (ports_range_start == ports_range_end)) {
+        /* non zero single value port */
+        retry = 0;
+    } else {
+        /* port value is zero or a range of ports is given */
+        retry = 1;
     }
 
+    rand_seed = ucs_generate_uuid((uintptr_t)saddr);
+
     do {
+        if ((ports_range_start != 0) || (ports_range_end != 0)) {
+            port = (uint16_t)((rand_r(&rand_seed) %
+                              (ports_range_end -
+                               ports_range_start + 1)) +
+                              ports_range_start);
+        } else {
+            port = 0;   /* use a random port */
```

Comment:
/* let the operating system choose the port */
it's random in other case as well, difference is who does the random